slatedb-java/src/main/java/com/slatedb/config/CompactorOptions.java

```diff
@@ -0,0 +1,237 @@
+package com.slatedb.config;
+
+import java.time.Duration;
+import java.util.Objects;
+
+/**
+ * Configuration options for the SlateDB background compactor.
+ * 
+ * The compactor is responsible for merging and organizing SSTables
+ * to maintain optimal read performance and storage efficiency.
+ */
+public final class CompactorOptions {
+    private final Duration pollInterval;
+    private final Duration manifestUpdateTimeout;
+    private final Long maxSstSizeBytes;
+    private final Integer maxConcurrentCompactions;
+    
+    private CompactorOptions(Builder builder) {
+        this.pollInterval = builder.pollInterval;
+        this.manifestUpdateTimeout = builder.manifestUpdateTimeout;
+        this.maxSstSizeBytes = builder.maxSstSizeBytes;
+        this.maxConcurrentCompactions = builder.maxConcurrentCompactions;
+    }
+    
+    /**
+     * Gets how often to check for compaction opportunities.
+     * 
+     * @return the poll interval, or null if using default
+     */
+    public Duration getPollInterval() {
+        return pollInterval;
+    }
+    
+    /**
+     * Gets the timeout for manifest update operations.
+     * 
+     * @return the manifest update timeout, or null if using default
+     */
+    public Duration getManifestUpdateTimeout() {
+        return manifestUpdateTimeout;
+    }
+    
+    /**
+     * Gets the target size for output SSTables during compaction.
+     * 
+     * @return the max SST size in bytes, or null if using default
+     */
+    public Long getMaxSstSizeBytes() {
+        return maxSstSizeBytes;
+    }
+    
+    /**
+     * Gets the maximum number of concurrent compaction jobs.
+     * 
+     * @return the max concurrent compactions, or null if using default
+     */
+    public Integer getMaxConcurrentCompactions() {
+        return maxConcurrentCompactions;
+    }
+    
+    /**
+     * Creates a new builder for CompactorOptions.
+     * 
+     * @return a new builder instance
+     */
+    public static Builder builder() {
+        return new Builder();
+    }
+    
+    /**
+     * Creates a new builder initialized with values from this configuration.
+     * 
+     * @return a new builder instance
+     */
+    public Builder toBuilder() {
+        return new Builder()
+                .pollInterval(pollInterval)
+                .manifestUpdateTimeout(manifestUpdateTimeout)
+                .maxSstSizeBytes(maxSstSizeBytes)
+                .maxConcurrentCompactions(maxConcurrentCompactions);
+    }
+    
+    /**
+     * Creates CompactorOptions with default settings.
+     * 
+     * @return CompactorOptions with all defaults
+     */
+    public static CompactorOptions defaultOptions() {
+        return builder().build();
+    }
+    
+    /**
+     * Creates CompactorOptions optimized for high throughput.
+     * 
+     * @return CompactorOptions for high throughput
+     */
+    public static CompactorOptions highThroughput() {
```

Comment:
Please remove highThroughput/lowResource. Similar reasoning to my comment below.